shared/hooks/useRelativeTime.tsx

```diff
@@ -0,0 +1,76 @@
+import { useEffect, useState } from 'react';
+
+import { formatDistanceToNowStrict } from 'date-fns';
```

Comment:
The implementation of the relative date format is quite simple using the built-in `Intl` object. A third-party library is simply not needed. Here is a [link to a simple example](https://blog.webdevsimplified.com/2020-07/relative-time-format/) of how to write something similar.